llvm/lib/Analysis/InstructionSimplify.cpp

```diff
@@ -3103,6 +3103,54 @@ static Value *simplifyICmpWithConstant(CmpInst::Predicate Pred, Value *LHS,
   return nullptr;
 }
 
+/// Get values V_i such that V uge V_i (Greater) or V ule V_i (!Greater).
+static void getUnsignedMonotonicValues(SmallPtrSetImpl<Value *> &Res, Value *V,
+                                       bool Greater, unsigned Depth = 0) {
```

Comment:
Would replace greater with an enum that also support `GreaterOrEq` and `LessOrEq`. Also think a bool for `Signed` makes sense.